### ISSUE TITLE
Several errors in livesplit causing timer stuttering and performance issues with game memory being read by autosplitters in several different games.

### ISSUE THREAD

**Issue by qusteka** [LiveSplit/LiveSplit#1144]
Here are the errors produced by windows:

Error 1:
### inb4 desktop complaint i've tried several different install locations; same result

Could not load file or assembly 'file:///F:\Users\OMITTED\Desktop\split\Components\Noesis.Javascript.dll' or one of its dependencies. An attempt was made to load a program with an incorrect format.

   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.UnsafeLoadFrom(String assemblyFile)
   at LiveSplit.UI.Components.ComponentManager.LoadFactory(String path)
   
Error 2:
   
The handle is invalid

   at LiveSplit.ComponentUtil.ExtensionMethods.ModulesWow64Safe(Process p)
   at LiveSplit.ComponentUtil.DeepPointer.DerefOffsets(Process process, IntPtr& ptr)
   at LiveSplit.ComponentUtil.DeepPointer.Deref[T](Process process, T& value)
   at LiveSplit.ASL.ASLState.GetValue(Process p, String type, DeepPointer pointer)
   at LiveSplit.ASL.ASLState.RefreshValues(Process p)
   at LiveSplit.ASL.ASLScript.DoUpdate(LiveSplitState state)
   at LiveSplit.UI.Components.ASLComponent.UpdateScript()
   
Error 3: 

HRESULT: [0x8007001E], Module: [SharpDX.DirectInput], ApiCode: [DIERR_INPUTLOST/InputLost], Message: The system cannot read from the specified device.

   at SharpDX.Result.CheckError()
   at SharpDX.DirectInput.Device.GetDeviceData(Int32 arg0, IntPtr arg1, Int32& arg2, Int32 arg3)
   at SharpDX.DirectInput.CustomDevice`3.GetBufferedData()
   at LiveSplit.Model.Input.GamepadHook.Poll()

**Comment by Iguaan** [comment]
I'm currently experiencing the same issue, have you found a solution or a workaround to this?

**Comment by qusteka** [comment]
Here is a short list of things that I did which alleviated the problem to varying degrees:
**It's not fixed but the performance hit is smaller and the issues occur less frequently**
1: Replaced `Noesis.Javascript.dll` with the one taken directly from the javascript dev kit file naming is case sensitive.

2: Created a new `*.asl` file where applicable manually copy pasting the code.

3: Asked the creators of the autsosplitters to optimize their splitters/had other people optimize them further.

4: Replaced the automatically downloaded autosplitters with the optimized ones via a `Scriptable Autosplitter Component` in the layout.

5: Making sure that the `SAC` is the last in the layout seems to make a difference.

6: Using different video capturing software than OBS/only using the absolute necessary in terms of scenes on OBS. 

7: Updating OBS

8: Not capturing livesplit by external software.

9: Disabling nvidia instant replay/capture/overlay

10: Disabling steam overlay.

That's all.

Issues still happen albeit not as often and if they do they have a much less severe impact.
